### PR TITLE
Merge pull request #240 from krlmlr/f-examine-to-check

### LINKED ISSUE #238: Remove some examine_ functions
because they throw an error.

If a function returns invisibly and throws an error, it should be named `check_*()`.

- [x] `check_cardinality_*_*()` (but keep `examine_cardinality()`)
- [x] `check_if_subset()` (inconsistent name, maybe `check_subset()`)?
- [x] `check_key()`

### PR DESCRIPTION
**Pull request by TSchiefer**
change some of the function names and update naming vignette

closes #238

### CODE REVIEW

File: vignettes/wip/dm-function-naming-logic.Rmd [hunk 1]
```diff
@@ -18,12 +18,9 @@ A `zoomed_dm` behaves in most situations like a table and there are mostly no ex
 1. `dm_`: There are two cases for this prefix:
     - The function takes a `dm` object as a first argument
     - The function constructs a `dm` object from other objects
-1. `examine_`: There are two cases for this prefix:
-    - The function throws an error if the respective check is not passed and otherwise returns the first argument invisibly.
-    - The function executes several checks and returns the results in tibble-form. 
+1. `check_`: Certain tests on the arguments are performed and an error is thrown in case they are not passed. If the tests are passed, the first argument is returned invisibly.
+1. `examine_`: The function executes checks (of which some may lead to an error if not passed) and returns an object containing an overview of the results of the checks.
```
Comment (krlmlr) [inline vignettes/wip/dm-function-naming-logic.Rmd]:
```suggestion
1. `examine_`: The function executes checks and returns an object containing an overview of the results of the checks. It should never throw an error.
```

File: R/check-cardinalities.R [hunk 2]
```diff
@@ -147,8 +154,23 @@ examine_cardinality <- function(parent_table, pk_column, child_table, fk_column)
...
+        "Cannot examine cardinality: Column(s) {tick(commas(as_string(pkc)))} not ",
+        "a unique key of {tick('parent_table')}."
+      )
+    )
+  }
+
+  if (!is_subset(!!ct, !!fkc, !!pt, !!pkc)) {
+    return(
+      glue(
+        "Cannot examine cardinality: Column(s) {tick(commas(as_string(fkc)))} of {tick('child_table')} not ",
```
Comment (krlmlr) [inline R/check-cardinalities.R]:
```suggestion
        "Column(s) {tick(commas(as_string(fkc)))} of {tick('child_table')} not ",
```

File: pkgdown/_pkgdown.yml [hunk 3]
```diff
@@ -94,14 +94,14 @@ reference:
 - title: Check keys and cardinalities
   contents:
   - dm_examine_constraints
-  - examine_key
-  - examine_if_subset
-  - examine_set_equality
+  - check_key
+  - check_subset
+  - check_set_equality
   - examine_cardinality
```
Comment (krlmlr) [inline pkgdown/_pkgdown.yml]: Can you please move up, below `dm_examine_constraints` ?

File: vignettes/wip/dm-function-naming-logic.Rmd (hunk 1 above)
Comment (krlmlr) [reply vignettes/wip/dm-function-naming-logic.Rmd]: Do we still need to rephrase here?

File: R/check-cardinalities.R [hunk 4]
```diff
@@ -17,97 +19,102 @@
...
 #' @param child_table Data frame.
 #' @param fk_column Column of `child_table` that has to be a foreign key to `pk_column` in `parent_table`.
 #'
 #' @name examine_cardinality
 #'
-#' @return Functions return `parent_table`, invisibly, if the check is passed, to support pipes.
+#' @return For `check_cardinality_?_?()`: Functions return `parent_table`, invisibly, if the check is passed, to support pipes.
 #' Otherwise an error is thrown and the reason for it is explained.
 #'
+#' For `examine_cardinality()`: Returns a character variable specifying the type of relationship between the two columns.
```
[the remark this answers is not in the document]
Comment (TSchiefer) [reply R/check-cardinalities.R]: created issue: https://github.com/krlmlr/dm/issues/246

**Comment by krlmlr** [comment]
Thanks!

**Pull request merged**